### PR TITLE
apps sc & wc: opensearch 2.6.0

### LINKED ISSUE #1022: [5] Upgrade Opensearch to v2.0.0
**What Chart should be upgraded?**

upstream/opensearch

**Why should this Chart be upgraded?**

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

**Acceptance criteria**

- I checked the migration of the new Chart:
    - [x] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

Make sure you also include this feature in this upgrade: https://github.com/elastisys/compliantkubernetes-apps/issues/1052

### PR DESCRIPTION
**Pull request by robinelastisys**
**What this PR does / why we need it**:
We want to have the newest version of OpenSearch to keep our package updated and great! This PR is created to adapt our cluster to go from using OpenSearch 1.3.4 to 2.5.1.

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #1022 

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [x] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [x] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/50-applications.yaml [hunk 1]
```diff
@@ -362,7 +362,7 @@ releases:
     app: opensearch-dashboards
     group: opensearch
   chart: ./upstream/opensearch-dashboards
-  version: 1.7.4
```
Comment (Xartos) [inline helmfile/50-applications.yaml:365]: Why is this `2.4.0` and not `2.5.0`?

File: helmfile/values/opensearch/dashboards.yaml.gotmpl [hunk 2]
```diff
@@ -54,6 +54,9 @@ ingress:
     - host: {{ .Values.opensearch.dashboards.subdomain }}.{{ .Values.global.baseDomain }}
       paths:
         - path: /
+          backend:
+            serviceName:
+            servicePort:
```
Comment (Xartos) [inline helmfile/values/opensearch/dashboards.yaml.gotmpl:59]: Why is this needed? Seems like it's incomplete?

File: helmfile/50-applications.yaml (hunk 1 above)
Comment (Xartos) [reply helmfile/50-applications.yaml:365]: For some reason the code preview shows the wrong diff. But I ofc meant the other "side"

File: helmfile/values/opensearch/dashboards.yaml.gotmpl (hunk 2 above)
Comment (robinelastisys) [reply helmfile/values/opensearch/dashboards.yaml.gotmpl:59]: Was needed to make Helmfile happy!

File: helmfile/50-applications.yaml (hunk 1 above)
Comment (robinelastisys) [reply helmfile/50-applications.yaml:365]: https://github.com/opensearch-project/helm-charts/blob/main/charts/opensearch-dashboards/Chart.yaml

Because OpenSearch and OpenSearch-Dashboard don't have the same version number. Not super logical but!

File: helmfile/values/opensearch/dashboards.yaml.gotmpl (hunk 2 above)
Comment (aarnq) [reply helmfile/values/opensearch/dashboards.yaml.gotmpl:59]: I'd prefer if you actually set them to an empty string if possible so it looks intentional, rather than like something is missing.

File: helmfile/values/opensearch/dashboards.yaml.gotmpl [hunk 3]
```diff
@@ -54,6 +54,9 @@ ingress:
     - host: {{ .Values.opensearch.dashboards.subdomain }}.{{ .Values.global.baseDomain }}
       paths:
         - path: /
+          backend:
+            serviceName: ""
+            servicePort: ""
```
Comment (Xartos) [inline helmfile/values/opensearch/dashboards.yaml.gotmpl:59]: I still think that this is strange, shouldn't you need to set the correct service name and port here? Or are you sure it works without it?

Comment (robinelastisys) [reply helmfile/values/opensearch/dashboards.yaml.gotmpl:59]: it seems to be working without it indeed. I had to double-check. Looks as if they made it work out of the default value rather than needing set ones: https://github.com/opensearch-project/helm-charts/blob/main/charts/opensearch-dashboards/values.yaml

**Comment by robinelastisys** [comment]
Needs more work as we discovered a bunch of issues when we turned over a rock or 2!

File: helmfile/charts/opensearch/securityadmin/templates/job.yaml [hunk 4]
```diff
@@ -50,6 +50,7 @@ spec:
             - /etc/cert/admin/tls.crt
             - -key
             - /etc/cert/admin/tls.key
+
```
Comment (aarnq) [inline helmfile/charts/opensearch/securityadmin/templates/job.yaml:53]: This line can be removed.

**Review by aarnq — COMMENTED** [review]
You also need to change the backup and slm job to use https w/ insecure.

File: helmfile/values/opensearch/dashboards.yaml.gotmpl (hunk 3 above)
Comment (OlleLarsson) [reply helmfile/values/opensearch/dashboards.yaml.gotmpl:59]: Make a note on why they are necessary to have specified.

**Comment by davidumea** [comment]
https://github.com/elastisys/compliantkubernetes-apps/issues/1154 Is this a problem in opensearch `v2.5.1`?

**Comment by aarnq** [comment]
> #1154 Is this a problem in opensearch `v2.5.1`?

Yes, seems like so.

**Review by Xartos — APPROVED** [review]

**Comment by viktor-f** [comment]
Did you test to do a backup restore as well? I'm not sure what changed in the major version, but if it changed the backup/restore apis then it would be good to test that and update docs if needed.

**Comment by robinelastisys** [comment]
> backup restore

Can confirm that this works without issues!

**Comment by robinelastisys** [comment]
Dashboard fix: Exported all objects in OpenSearch 2.0 to make them have the right syntax.

**Comment by OlleLarsson** [comment]
fixes #1052

**Review by viktor-f — COMMENTED** [review]
No migration steps needed for this?
Otherwise I think it looks good, although I have not tested it.

File: WIP-CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,7 @@
 
 - The ThanosReceiveHttpRequestErrorRateHigh, OpenSearchTooFewNodesRunning, and MetricsFromScClusterIsMissing prometheus-alerts were optimized to not be triggered by kured and to reduce other unwanted behaviors.
 - Support section of the grafana/opensearch welcome dashboards is removed by default, you can now add custom text at the end of the dashboards via the sc-config.
+- OpenSearch was upgraded from version "1.3.4" to "2.5.1. Our fluentd-plugins was also changed from using "fluent-plugin-opensearch: 1.0.2" to "fluent-plugin-opensearch: 1.0.8".
```
Comment (OlleLarsson) [inline WIP-CHANGELOG.md:15]: I would suggest noting the fluentd changes as a separate entry.
```suggestion
- OpenSearch has been upgraded from version `v1.3.4` to `v2.5.1`. Our fluentd-plugins was also changed from using "fluent-plugin-opensearch: 1.0.2" to "fluent-plugin-opensearch: 1.0.8".
```

**Comment by OlleLarsson** [comment]
Os and Osd 2.3 was released today
https://github.com/opensearch-project/opensearch-build/releases/tag/2.3.0

**Comment by robinelastisys** [comment]
> Opensearch 2.3 was released today
> 
> * https://github.com/opensearch-project/opensearch-build/releases/tag/2.3.0
> * https://github.com/opensearch-project/helm-charts/releases/tag/opensearch-dashboards-2.5.0

Awesome! Will try to upgrade then!

**Comment by robinelastisys** [comment]
Now it have all been checked! I think its ready to be merged!

**Review by raviranjanelastisys — APPROVED** [review]
